### PR TITLE
bootspec: Add PE file name to log messages

### PR DESCRIPTION
**Pull request by medhefgo**
Fixes: #23783

### CODE REVIEW

**Comment by yuwata** [comment]
Hm? How does this fix the issue??

**Comment by medhefgo** [comment]
It allows identifying the cause for the error messages from `find_sections`. I'm too lazy to change all those error messages to contain a path name. :D

**Comment by yuwata** [comment]
I see. Let me confirm that, IIUC the thread in the issue page, it is not an issue, is it? As we ignore the wrong entry.

I think that may be boring, but it is better to
- include the path in the log messages in `find_sections()`,
- downgrade log level from LOG_ERR -> LOG_WARNING, as it is not critical,
- and change "refusing" -> "ignoring", as the errors are ignored.

**Comment by keszybz** [comment]
> * downgrade log level from LOG_ERR -> LOG_WARNING, as it is not critical,
> 
>  * and change "refusing" -> "ignoring", as the errors are ignored.

I think we want this too.

File: src/shared/bootspec.c [hunk 1]
```diff
@@ -557,28 +558,28 @@ static int find_sections(
         struct DosFileHeader dos;
         n = pread(fd, &dos, sizeof(dos), 0);
         if (n < 0)
-                return log_error_errno(errno, "Failed read DOS header: %m");
+                return log_warning_errno(errno, "%s: Failed read DOS header: %m", path);
```
Comment (yuwata) [inline src/shared/bootspec.c]: "Failed _to_ read". And missing ", ignoring".

File: src/shared/bootspec.c [hunk 2]
```diff
@@ -557,28 +558,28 @@ static int find_sections(
...
-                return log_error_errno(SYNTHETIC_ERRNO(EBADMSG), "DOS executable magic missing, refusing.");
+                return log_warning_errno(SYNTHETIC_ERRNO(EBADMSG), "%s: DOS executable magic missing, ignoring.", path);
 
         uint64_t start = unaligned_read_le32(&dos.ExeHeader);
 
         struct PeHeader pe;
         n = pread(fd, &pe, sizeof(pe), start);
         if (n < 0)
-                return log_error_errno(errno, "Failed to read PE header: %m");
+                return log_warning_errno(errno, "%s: Failed to read PE header: %m", path);
```
Comment (yuwata) [inline src/shared/bootspec.c]: missing ", ignoring"

File: src/shared/bootspec.c [hunk 3]
```diff
@@ -588,9 +589,9 @@ static int find_sections(
                   n_sections * sizeof(struct PeSectionHeader),
                   start + sizeof(pe) + unaligned_read_le16(&pe.FileHeader.SizeOfOptionalHeader));
         if (n < 0)
-                return log_error_errno(errno, "Failed to read section data: %m");
+                return log_warning_errno(errno, "%s: Failed to read section data: %m", path);
```
Comment (yuwata) [inline src/shared/bootspec.c]: missing ", ignoring"

File: src/shared/bootspec.c [hunk 4]
```diff
@@ -605,34 +606,34 @@ static int find_sections(
...
+                        return log_warning_errno(SYNTHETIC_ERRNO(EBADMSG), "%s: Section %s too large, ignoring.", path, sections[i].Name);
 
                 k = new(char, size+1);
                 if (!k)
                         return log_oom();
 
                 n = pread(fd, k, size, offset);
                 if (n < 0)
-                        return log_error_errno(errno, "Failed to read section payload: %m");
+                        return log_warning_errno(errno, "%s: Failed to read section payload: %m", path);
```
Comment (yuwata) [inline src/shared/bootspec.c]: missing ", ignoring"

File: src/shared/bootspec.c [hunk 5]
```diff
@@ -605,34 +606,34 @@ static int find_sections(
...
                 if ((size_t) n != size)
-                        return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read while reading section payload, refusing:");
+                        return log_warning_errno(SYNTHETIC_ERRNO(EIO), "%s: Short read while reading section payload, ignoring:", path);
 
                 /* Allow one trailing NUL byte, but nothing more. */
                 if (size > 0 && memchr(k, 0, size - 1))
-                        return log_error_errno(SYNTHETIC_ERRNO(EBADMSG), "Section contains embedded NUL byte: %m");
+                        return log_warning_errno(SYNTHETIC_ERRNO(EBADMSG), "%s: Section contains embedded NUL byte.", path);
```
Comment (yuwata) [inline src/shared/bootspec.c]: missing ", ignoring"

**Review by yuwata — COMMENTED** [review]
Below are not your fault, but please address them.

**Pull request merged**